install/src/HieInstall.hs

```diff
@@ -115,13 +114,11 @@ defaultMain = do
     phony "cabal-build"  (need ["build-data", "cabal-build-latest"])
     phony "cabal-build-data" $ do
       need ["submodules"]
-      need ["cabal"]
       cabalBuildData
     forM_
       ghcVersions
       (\version -> phony ("cabal-hie-" ++ version) $ do
         need ["submodules"]
-        need ["cabal"]
```

Comment:
If we remove the cabal target we should remove all `cabal-*` targets from stack executions and all `stack-*` targets from cabal executions

Reply:
I think that is an unrelated change. The `cabal` target is for checking an existing installation of cabal. The `cabal-*` and `stack-*` targets can be removed in another PR.

Reply:
Fine with make it in other pr, but we have to take in account that  `stack install.hs cabal-hie-8.6.5` will not check the cabal version and will not install them if it is not present.